contracts/ERC677BridgeTokenRewardable.sol

```diff
@@ -0,0 +1,66 @@
+pragma solidity 0.4.24;
+
+import "./ERC677BridgeToken.sol";
+
+
+contract ERC677BridgeTokenRewardable is ERC677BridgeToken {
+
+    address public blockRewardContract;
+    address public validatorSetContract;
+
+    constructor(
+        string _name,
+        string _symbol,
+        uint8 _decimals
+    ) public ERC677BridgeToken(_name, _symbol, _decimals) {}
+
+    function setBlockRewardContract(address _blockRewardContract) onlyOwner public {
+        require(_blockRewardContract != address(0) && isContract(_blockRewardContract));
+        blockRewardContract = _blockRewardContract;
+    }
+
+    function setValidatorSetContract(address _validatorSetContract) onlyOwner public {
+        require(_validatorSetContract != address(0) && isContract(_validatorSetContract));
+        validatorSetContract = _validatorSetContract;
+    }
+
+    modifier onlyBlockRewardContract() {
+        require(msg.sender == blockRewardContract);
+        _;
+    }
+
+    modifier onlyValidatorSetContract() {
+        require(msg.sender == validatorSetContract);
+        _;
+    }
+
+    function mintReward(address[] _receivers, uint256[] _rewards) external onlyBlockRewardContract {
+        for (uint256 i = 0; i < _receivers.length; i++) {
+            address to = _receivers[i];
+            uint256 amount = _rewards[i];
+
+            // Mint `amount` for `to`
+            totalSupply_ = totalSupply_.add(amount);
```

Comment:
may we use `super.mint(to, amount)` here?

Reply:
No, we can't use this with the current version of `openzeppelin-solidity` package because `super.mint` has the next modifiers:

![image](https://user-images.githubusercontent.com/33550681/50452254-0bb2ae00-094a-11e9-88a9-ed6261d31928.png)

(see `flats/ERC677BridgeToken_flat.sol`)

However, we could use [`_mint` internal function](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/7361ffd26b6354499d0b68d6d2d9da1f8cb21c8b/contracts/token/ERC20/ERC20.sol#L151) which is available in the last version of `openzeppelin-solidity`, but npm package update is required in this case.